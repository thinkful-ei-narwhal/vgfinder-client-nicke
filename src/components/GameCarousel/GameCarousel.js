import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./GameCarousel.css";

export default class GamesCarousel extends Component {
  state = {
    currentSlide: 0,
    loaded: true,
  };

  componentDidMount() {
    if (!this.props.reel) {
      throw new Error("Supply an image reel");
    }
    this.setState({
      reel: this.props.reel,
      currentSlide: 0,
      loaded: true,
    });
  }

  plusSlide() {
    this.setState({ currentSlide: this.state.currentSlide + 1 });
    if (!this.props.isSingleGame)
      this.props.setActiveGame(
        this.props.reel.find(
          (reelObj, index) => this.state.currentSlide + 1 === index
        ).gameId
      );
  }

  minusSlide() {
    this.setState({ currentSlide: this.state.currentSlide - 1 });
    if (!this.props.isSingleGame)
      this.props.setActiveGame(
        this.props.reel.find(
          (reelObj, index) => this.state.currentSlide - 1 === index
        ).gameId
      );
  }

  mapImages() {
    return this.props.reel.map((reelObj, index) => {
      const slide =
        this.state.currentSlide === index ? "showSlide" : "mySlides";
      return this.props.isSingleGame ? (
        <div key={index} className={`${slide} fade`}>
          <div className="numbertext">
            {index + 1} / {this.props.reel.length}
          </div>
          <img src={reelObj.imgUrl} alt={`${reelObj.title} thumbnail`} />
        </div>
      ) : (
        <div key={index} className={`${slide} fade`}>
          <div className="numbertext">
            {index + 1} / {this.props.reel.length}
          </div>
          <Link to={`/games/${reelObj.gameId}`}>
            {" "}
            <img src={reelObj.imgUrl} alt={`${reelObj.title} thumbnail`} />
          </Link>
        </div>
      );
    });
  }

  mapDots() {
    return this.props.reel.map((reelObj, index) => {
      return this.props.isSingleGame ? (
        <span
          key={index}
          className="dot"
          onClick={() => this.setState({ currentSlide: index })}
        ></span>
      ) : (
        <span
          key={index}
          className="dot"
          onClick={() => {
            this.setState({ currentSlide: index });
            this.props.setActiveGame(reelObj.gameId);
          }}
        ></span>
      );
    });
  }

  renderOnLoad() {
    return (
      <>
        <div className="slideshow-container">
          {this.mapImages()}
          {this.state.currentSlide === 0 ? null : (
            <a href="#0" className="prev" onClick={() => this.minusSlide()}>
              &#10094;
            </a>
          )}
          {this.state.currentSlide === this.props.reel.length - 1 ? null : (
            <a href="#0" className="next" onClick={() => this.plusSlide()}>
              &#10095;
            </a>
          )}
        </div>
        <br />
        <div className="style-align">{this.mapDots()}</div>
      </>
    );
  }

  render() {
    return <>{this.state.loaded ? this.renderOnLoad() : null}</>;
  }
}
