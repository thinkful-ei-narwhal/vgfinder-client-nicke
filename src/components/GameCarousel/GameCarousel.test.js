import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import GameCarousel from './GameCarousel'

describe(`GameCarousel component`, () => {
  const propsGamesHomePage = {
    reel: [
      {
        imgUrl: "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
        gameId: "2"
      },
      {
        imgUrl: "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
        gameId: "2"
      }
    ],
    isSingleGame: false,
    setActiveGame: 2
  }

  const propsGamePage = {
    isSingleGame: true,
    reel: [{
      imgUrl: "https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500",
      gameId: "2"
    }]
  }

  it('renders a GameCarousel by default', () => {
    const wrapper = shallow(<GameCarousel />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the NotefulForm given GamesHomePage props', () => {
    const wrapper = shallow(<GameCarousel {...propsGamesHomePage} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the NotefulForm given GamePage props', () => {
    const wrapper = shallow(<GameCarousel {...propsGamePage} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
