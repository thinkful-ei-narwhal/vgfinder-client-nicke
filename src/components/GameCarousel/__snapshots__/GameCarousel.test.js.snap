// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCarousel component renders the GameCarousel given GamePage props 1`] = `
<Fragment>
  <div
    className="slideshow-container"
  >
    <div
      className="showSlide fade container"
      key="0"
      style={
        Object {
          "backgroundColor": "black",
          "backgroundImage": "url(https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500)",
          "backgroundPosition": "center",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "contain",
        }
      }
    >
      <div
        className="numbertext "
      >
        1
         / 
        1
      </div>
    </div>
  </div>
  <br />
  <div
    className="style-align"
  >
    <span
      className="dot"
      key="0"
      onClick={[Function]}
    />
  </div>
</Fragment>
`;

exports[`GameCarousel component renders the GameCarousel given GamesHomePage props 1`] = `
<Fragment>
  <div
    className="slideshow-container"
  >
    <Link
      key="0"
      to="/games/2"
    >
      <div
        className="showSlide fade container"
        style={
          Object {
            "backgroundColor": "black",
            "backgroundImage": "url(https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500)",
            "backgroundPosition": "center",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "contain",
          }
        }
      >
        <div
          className="numbertext"
        >
          1
           / 
          2
        </div>
      </div>
    </Link>
    <Link
      key="1"
      to="/games/3"
    >
      <div
        className="mySlides fade container"
        style={
          Object {
            "backgroundColor": "black",
            "backgroundImage": "url(https://steamcdn-a.akamaihd.net/steam/apps/367520/capsule_616x353.jpg?t=1577747500)",
            "backgroundPosition": "center",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "contain",
          }
        }
      >
        <div
          className="numbertext"
        >
          2
           / 
          2
        </div>
      </div>
    </Link>
    <a
      className="next game-carousel-button"
      href="#0"
      onClick={[Function]}
    >
      ‚ùØ
    </a>
  </div>
  <br />
  <div
    className="style-align"
  >
    <span
      className="dot"
      key="0"
      onClick={[Function]}
    />
    <span
      className="dot"
      key="1"
      onClick={[Function]}
    />
  </div>
</Fragment>
`;
